frontend/src/_services/authentication.service.ts

```diff
@@ -99,4 +103,35 @@ export class AuthenticationService extends ApiService {
             }),
         );
     }
+    public startTokenExpiryCheck(): void {
+        const tokenExpirationTime = this.getTokenExpiration();
+
+        if (!tokenExpirationTime) {
+            this.logout();
+            return;
+        }
+
+        const oneMinute = 60 * 1000;
+        const checkInterval = oneMinute;
+
+        setInterval(() => {
+            const now = Date.now();
+            const timeLeft = tokenExpirationTime - now;
+
+            if (timeLeft <= 0) {
+                alert('Your session has expired. Please log in again.');
+                this.logout();
+            } else if (timeLeft < 5 * oneMinute) {
+                this.toastService.showToast('Your session will expire in less than 5 minutes. Please log in again');
+            }
+        }, checkInterval);
+    }
+
+    private getTokenExpiration() {
+        const user = this.userValue;
+        if (!user || !user.accessToken) return null;
+
+        const tokenPayload = JSON.parse(atob(user.accessToken.split('.')[1]));
+        return tokenPayload.exp ? tokenPayload.exp * 1000 : null;
```

Comment:
Think it would be easier to use `jwt-decode` library but I guess this is still pretty readable haha